libsnowflakeclient/include/snowflake_client.h

```diff
@@ -68,11 +70,18 @@ typedef enum sf_status {
  */
 typedef enum sf_error_code {
     SF_NO_ERROR,
-    SF_OUT_OF_MEMORY,
-    SF_REQUEST_TIMEOUT,
-    SF_BAD_REQUEST,
-    SF_DATA_CONVERSION,
-    SF_BAD_DATA_OUTPUT_TYPE
+    SF_ERROR_OUT_OF_MEMORY,
+    SF_ERROR_REQUEST_TIMEOUT,
+    SF_ERROR_DATA_CONVERSION,
+    SF_ERROR_BAD_DATA_OUTPUT_TYPE,
+    SF_ERROR_BAD_CONNECTION_PARAMS,
+    SF_ERROR_STRING_FORMATTING,
+    SF_ERROR_STRING_COPY,
+    SF_ERROR_BAD_REQUEST,
+    SF_ERROR_BAD_JSON,
+    SF_ERROR_RETRY,
+    SF_ERROR_CURL,
+    SF_ERROR_BAD_ATTRIBUTE_TYPE
```

Comment:
Yea I'm okay with either. I think we should keep the type names as SNOWFLAKE_ and change all the enum values as SF_ since some of the enum names can get quite long.